File: scripts/print_utils.dart
import 'dart:io';

// ANSI 颜色代码 - 与 Python print_utils.py 保持一致
class Colors {
  // 基础颜色
  static const String reset = '\x1B[0m';
  static const String red = '\x1B[31m';
  static const String green = '\x1B[32m';
  static const String yellow = '\x1B[33m';
  static const String blue = '\x1B[34m';
  static const String magenta = '\x1B[35m';
  static const String cyan = '\x1B[36m';
  static const String white = '\x1B[37m';

  // 样式
  static const String bold = '\x1B[1m';
  static const String dim = '\x1B[2m';
  static const String italic = '\x1B[3m';
  static const String underline = '\x1B[4m';
}

// 彩色输出函数 - 与 Python 版本保持一致
void printStep(String step, String message) {
  stdout.writeln('\n');
  stdout.writeln('++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++');
  stdout.writeln('${Colors.cyan}[$step]${Colors.reset} $message');
}

void printInfo(String message) {
  stdout.writeln('${Colors.blue}[INFO]${Colors.reset} $message');
}

void printSuccess(String message) {
  stdout.writeln('${Colors.green}[SUCCESS]${Colors.reset} $message');
}

void printError(String message) {
  stdout.writeln('${Colors.red}[ERROR]${Colors.reset} $message');
}

void printWarning(String message) {
  stdout.writeln('${Colors.yellow}[WARNING]${Colors.reset} $message');
}

// 额外的实用函数
void printBold(String message) {
  stdout.writeln('${Colors.bold}$message${Colors.reset}');
}

void printDim(String message) {
  stdout.writeln('${Colors.dim}$message${Colors.reset}');
}

void printCyan(String message) {
  stdout.writeln('${Colors.cyan}$message${Colors.reset}');
}

void printMagenta(String message) {
  stdout.writeln('${Colors.magenta}$message${Colors.reset}');
}

File: scripts/print_utils.py
import sys
import subprocess

def check_and_install_colorama():
    """检查并安装 colorama 包"""
    try:
        import colorama
    except ImportError:
        print("正在安装 colorama 包...")
        try:
            subprocess.check_call([sys.executable, "-m", "pip", "install", "colorama"])
            print("colorama 安装成功")
        except subprocess.CalledProcessError as e:
            print(f"安装 colorama 失败: {e}")
            sys.exit(1)

# 检查并安装 colorama
check_and_install_colorama()

# 现在可以安全地导入 colorama
from colorama import init, Fore, Style

# 初始化 colorama
init()

def print_step(step, message):
    """打印步骤信息"""
    print('\n')
    print(f"++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++")
    print(f"{Fore.CYAN}[{step}]{Style.RESET_ALL} {message}")

def print_info(message):
    """打印普通信息"""
    print(f"{Fore.BLUE}[INFO]{Style.RESET_ALL} {message}")

def print_success(message):
    """打印成功信息"""
    print(f"{Fore.GREEN}[SUCCESS]{Style.RESET_ALL} {message}")

def print_error(message):
    """打印错误信息"""
    print(f"{Fore.RED}[ERROR]{Style.RESET_ALL} {message}")

def print_warning(message):
    """打印警告信息"""
    print(f"{Fore.YELLOW}[WARNING]{Style.RESET_ALL} {message}") 